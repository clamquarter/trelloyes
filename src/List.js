import Card from './Card'
import React from 'react'
import './List.css'



function List(props){
    
    return ( <section className='List'>
    <header><h2>{props.header}List testing!</h2></header>
    <p>List test!
        <ul></ul>
    </p>
    <div className='List-cards'>
        {props.cards.map((card) => 
        <Card key={card.id}
        title={card.title}
        content={card.content}
    />)}

    <button
          type='button'
          className='List-add-button'
        >
            Add Card
    </button>
    </div>
    </section>
    )
        
}

export default List