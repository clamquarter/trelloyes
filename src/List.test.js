import React from 'react'
import ReactDOM from 'react-dom'
import List from './List'
import renderer from 'react-test-renderer'

it('renders without crash', () => {
    const div = document.createElement('div')

    ReactDOM.render(<List key='test'
        header='test'
        cards={[]}/>, div)
    ReactDOM.unmountComponentAtNode(div)
})

it('renders the UI as expected', () => {
    const tree = renderer
    .create(<List key='test'
    header='test'
    cards={[]}/>)
    .toJSON()
    expect(tree).toMatchSnapshot()

})