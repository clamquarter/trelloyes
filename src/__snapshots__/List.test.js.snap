// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <button
      className="List-add-button"
      type="button"
    >
      Add Card
    </button>
  </div>
</section>
`;
